src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -87,7 +87,84 @@ public IWebHostBuilder CreateWebHostBuilder(IConsole console, string[] urls, str
                         services.ConfigureMetrics(context.Configuration);
                     }
                 })
-                .UseUrls(urls)
+                .ConfigureKestrel( (context, options) =>
+                {
+                    //Note our priorities for hosting urls don't match the default behavior.
+                    //Default Kestrel behavior priority
+                    //1) ConfigureKestrel settings
+                    //2) Command line arguments (--urls)
+                    //3) Environment variables (ASPNETCORE_URLS, then DOTNETCORE_URLS)
+
+                    //Our precedence
+                    //1) Environment variables (ASPNETCORE_URLS, DotnetMonitor_Metrics__Endpoints)
+                    //2) Command line arguments (these have defaults) --urls, --metricUrls
+                    //3) ConfigureKestrel is used for fine control of the server, but honors the first two configurations.
+
+                    string hostingUrl = context.Configuration.GetValue<string>(WebHostDefaults.ServerUrlsKey);
+                    if (!string.IsNullOrEmpty(hostingUrl))
+                    {
+                        urls = ConfigurationHelper.SplitValue(hostingUrl);
+                    }
+
+                    if (metrics)
+                    {
+                        string metricUrlFromConfig = context.Configuration.GetValue<string>(ConfigurationHelper.MakeKey(MetricsOptions.ConfigurationKey, nameof(MetricsOptions.Endpoints)));
```

Comment:
Are you certain that GetValue will look deep into the configuration. You are asking it to find Metrics:Endpoints, but I'm not certain GetValue does traversal down the configuration graph.

Reply:
It does work, but I switched this over to use your binding suggestion.